### PR TITLE
Update `revm` to latest `main`

### PR DESCRIPTION
**Pull request by ngotchac**
## Motivation

I needed some new functionalities from `revm`'s latest `main` branch, so I figured I'd create a draft PR until there is a new release, or if you want to use the latest version from `git` this could be merged.

## Solution

The updates are mostly internal. Externally it's mostly about renaming `subroutine` to `journaled_state`, replacing the `Account::Filth` flag with the corresponding boolean getters, and replacing the execution results from a tuple into a struct.

### CODE REVIEW

**Comment by mattsse** [comment]
oh

looks like we where working on the same thing https://github.com/foundry-rs/foundry/pull/2877

**Comment by ngotchac** [comment]
Okay, closing this as duplicate of #2877 !

File: evm/src/executor/fork/cache.rs [hunk 1]
```diff
@@ -204,10 +204,11 @@ impl MemDb {
                 accounts.insert(add, acc.info);
 
                 let acc_storage = storage.entry(add).or_default();
-                if acc.filth.abandon_old_storage() {
+                if acc.storage_cleared {
```
Comment (mattsse) [inline evm/src/executor/fork/cache.rs:207]: this was the only thing I wasn't sure about whether this identical

the previous was:

```rust
impl Filth {
    pub fn abandon_old_storage(&self) -> bool {
        matches!(self, Self::Destroyed | Self::NewlyCreated)
    }
```

wdyt @rakita

Comment (rakita) [reply evm/src/executor/fork/cache.rs:207]: It seems correct: https://github.com/bluealloy/revm/blob/24622a4c9e450217f6a723008a75e3faa3b7cc5d/crates/revm/src/db/in_memory_db.rs#L126

**Pull request closed without merging**